File: dev/src/com/borschtlabs/gytm/dev/ManhattanDistanceHeuristic.java
package com.borschtlabs.gytm.dev;

import com.badlogic.gdx.ai.pfa.Heuristic;

/**
 * @author octopussy
 */
public class ManhattanDistanceHeuristic implements Heuristic<TurnArea.WayPoint> {

   @Override
   public float estimate(TurnArea.WayPoint node, TurnArea.WayPoint endNode) {
      return Math.abs(endNode.getX() - node.getX()) + Math.abs(endNode.getY() - node.getY());
   }
}

File: dev/src/com/borschtlabs/gytm/dev/ManhattanDistanceHeuristic.kt
package com.borschtlabs.gytm.dev

import com.badlogic.gdx.ai.pfa.Heuristic

/**
 * @author octopussy
 */
class ManhattanDistanceHeuristic : Heuristic<TurnArea.WayPoint> {

    override fun estimate(node: TurnArea.WayPoint, endNode: TurnArea.WayPoint): Float {
        val xx = (endNode.x - node.x).toDouble()
        val yy = (endNode.y - node.y).toDouble()
        val i = Math.abs(xx) + Math.abs(yy)
        //val i = Math.sqrt(xx * xx + yy * yy)

        return i.toFloat()
    }
}
